### ISSUE TITLE
--without selinux needed if casync built with -Dselinux=false

### ISSUE THREAD

**Issue by w-vi** [systemd/casync#125]
I have built casync without the selinux support and when I tried to run
`casync make mydir.caidx  mydir` it failed with `Failed to run synchronizer: Operation not supported` because it tries to store selinux labels. Fix is to run the command with `--without=selinux` but shouldn't it be that if built withoiut selinux then the `--without=selinux` is implied and not needed? I am happy to look into it and provide a patch, but asking before I dive into it.

**Comment by ejoerns** [comment]
Just read your issue and remembered I seem to have forgotten to make a PR for this patch on my stack:

``` diff
diff --git a/src/caformat.h b/src/caformat.h
index 1645cca..0168264 100644
--- a/src/caformat.h
+++ b/src/caformat.h
@@ -148,7 +148,9 @@ enum {
                 CA_FORMAT_WITH_SUBVOLUME_RO|
                 CA_FORMAT_WITH_XATTRS|
                 CA_FORMAT_WITH_ACL|
+#if HAVE_SELINUX
                 CA_FORMAT_WITH_SELINUX|
+#endif
                 CA_FORMAT_WITH_FCAPS,
 
         CA_FORMAT_WITH_UNIX = /* Conservative UNIX file properties */
@@ -195,7 +197,9 @@ enum {
                 CA_FORMAT_WITH_SUBVOLUME|
                 CA_FORMAT_WITH_SUBVOLUME_RO|
                 CA_FORMAT_WITH_ACL|
+#if HAVE_SELINUX
                 CA_FORMAT_WITH_SELINUX|
+#endif
                 CA_FORMAT_WITH_FCAPS,
 
         CA_FORMAT_WITH_FUSE = /* All bits that may also be exposed via fuse */
@@ -246,7 +250,9 @@ enum {
                 CA_FORMAT_WITH_SUBVOLUME_RO|
                 CA_FORMAT_WITH_XATTRS|
                 CA_FORMAT_WITH_ACL|
+#if HAVE_SELINUX
                 CA_FORMAT_WITH_SELINUX|
+#endif
                 CA_FORMAT_WITH_FCAPS,
 
         CA_FORMAT_DEFAULT = /* The default set of flags */
```

I assume this is what you encountered, too and assume this is a valid solution.
I don't want to take away the patch opportunity from you thus either feel free to post you changes or let me know if you want me to make a PR with my patch.

**Comment by w-vi** [comment]
@ejoerns Cool, in that case go ahead and for me it is one less thing to do :)

**Comment by ejoerns** [comment]
@w-vi ok, I've created #127 for that

**Comment by keszybz** [comment]
Closed by #127. @ejoerns in the future, add "Fixes #..." in the patch text, so github will close the issue automatically.
